### DISCUSSION TITLE
Using Khoj chat as chatbot for personal website

### DISCUSSION THREAD

**Discussion by Jaydon-chai** [khoj-ai/khoj#188] [Ideas]
Just thought of a cool way to use the Khoj Chat, is it possible at all to host the khoj chat with your notes on a server and embed it as a chatbot into a website?

Then people could see what your thoughts are on stuff without you actually needing to be there. Obviously would require care in not revealing personal information.

**Comment by debanjum** [comment]
@Jaydon-chai, cool idea!

It's very possible to host Khoj Chat on a server. Khoj Chat is already served as a webpage, by default at http://localhost:8000/chat. All you'd need to do is expose Khoj at your server's public I.P. For e.g by starting it with `khoj --host 0.0.0.0` and then access it at `http://name-of-your-website/chat`. 

You could also embed Khoj Chat within an existing website via an [iframe](https://www.w3schools.com/tags/tag_iframe.ASP) or just access the `/chat` API directly and style the chat page yourself however you'd want.

Of course, as you mentioned, you'd need to ensure personal information isn't revealed. You can do that by separating the files indexed by the Khoj that is running on your public server.
